src/language/server.ts

```diff
@@ -277,13 +278,69 @@ connection.onCompletionResolve(
 /**
  * Execute the entire playground script.
  */
-connection.onRequest('executeAll', async (params) => {
-  const connectionOptions = params.connectionOptions || {};
-  const runtime = new ElectronRuntime(
-    await CliServiceProvider.connect(params.connectionString, connectionOptions)
-  );
+connection.onRequest('executeAll', (params, token) => {
+  return new Promise((resolve) => {
+    // Use Node worker threads to isolate each run of a playground
+    // to be able to cancel evaluation of infinite loops.
+    //
+    // There is an issue with support for `.ts` files.
+    // Trying to run a `.ts` file in a worker thread results in error:
+    // `The worker script extension must be “.js” or “.mjs”. Received “.ts”`
+    // As a workaround require `.js` file from the out folder.
+    //
+    // TODO: After webpackifying the extension replace
+    // the workaround with some similar 3rd-party plugin
+    const worker = new WorkerThreads(path.resolve(__dirname, 'worker.js'), {
+      // The workerData parameter sends data to the created worker
+      workerData: {
+        codeToEvaluate: params.codeToEvaluate,
+        connectionString: params.connectionString,
+        connectionOptions: params.connectionOptions
+      }
+    });
+
+    // Listen for results from the worker thread
+    worker.on('message', ([error, result]) => {
+      if (error) {
+        connection.sendNotification(
+          'showErrorMessage',
+          error.message
+        );
+      }
+
+      worker.terminate(); // Close the worker thread
 
-  return await runtime.evaluate(params.codeToEvaluate);
+      return resolve(result);
+    });
+
+    // Listen for cancellation request from the language server client
+    token.onCancellationRequested(async () => {
+      connection.console.log('Playground cancellation requested');
+      connection.sendNotification(
+        'showInformationMessage',
+        'The long running playground operation was canceled'
```

Comment:
Maybe `The running playground operation was canceled.` - since it might not be a long running operation that is cancelled. Cancelling a playground that is running is super smooth.